### PR TITLE
refactored esent code to support project k and save project name along w...

### PR DESCRIPTION
**Pull request by heejaechang**
...ith project path. this is slightly different than saving context but should fulfill the need.

### CODE REVIEW

**Comment by heejaechang** [comment]
@srivatsn @mavasani @shyamnamboodiripad @tmeschter @jmarolf can you take a look? thank you

File: src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs [hunk 1]
```diff
@@ -365,48 +320,76 @@ private static void WriteToStream(Stream inputStream, Stream esentStream, Cancel
...
                 return false;
             }
 
-            return TryGetUniqueFileId(document.Project.FilePath, out projectId) && TryGetUniqueFileId(document.FilePath, out documentId);
+            key = new EsentStorage.Key(projectId, projectNameId);
+            return true;
+        }
+
+        private bool TryGetProjectId(Project project, out int projectId, out int projectNameId)
+        {
```
Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/Workspace/Esent/EsentPersistentStorage.cs]: basically this is the change. I added projectNameId to (projectId, documentId) tuple.

**Comment by heejaechang** [comment]
fixes #1860

**Comment by heejaechang** [comment]
ping?

**Comment by heejaechang** [comment]
@srivatsn @mavasani @shyamnamboodiripad @tmeschter @jmarolf ping?

**Comment by srivatsn** [comment]
:+1:

**Pull request merged**